### PR TITLE
Implement CI scripts for TPM 1.2 simulator and tcsd.

### PR DESCRIPTION
**Pull request by twitchy-jsonp**

### CODE REVIEW

File: ci/setup_tpm12_simulator.sh [hunk 1]
```diff
@@ -63,8 +63,46 @@ fetch_simulator () {
...
+  echo "${SIM_PID}" > "${BUILD_BASE}/sim_pid"
+  disown
+  sleep 2
+}
+
+setup_tpm () {
+  echo "Starting the TPM..."
+  ${SIMULATOR_SRC}/libtpm/utils/tpmbios -v
+  echo "Creating a fake EK..."
+  ${SIMULATOR_SRC}/libtpm/utils/createek || true
```
Comment (mjg59) [inline ci/setup_tpm12_simulator.sh]: Why || true? If this fails we presumably want to bail

File: ci/setup_tpm12_simulator.sh [hunk 2]
```diff
@@ -63,8 +63,46 @@ fetch_simulator () {
...
+  cd "${SIMULATOR_SRC}/libtpm" && make -j4
+}
+
+run_simulator () {
+  mkdir -pv "${BUILD_BASE}/NVRAM"
+  export TPM_PORT='6545'
+  export TPM_PATH="${BUILD_BASE}/NVRAM"
+  export TPM_SERVER_NAME='localhost'
+  export TPM_SERVER_PORT='6545'
+  export TCSD_TCP_DEVICE_PORT='6545'
```
Comment (mjg59) [inline ci/setup_tpm12_simulator.sh]: Should this be in run_tcsd? Possibly define 6545 as a constant elsewhere.

File: ci/setup_tpm12_simulator.sh (hunk 1 above)
Comment (twitchy-jsonp) [reply ci/setup_tpm12_simulator.sh]: This always fails with `Error TPM deactivated from TPM_CreateEndorsementKeyPair` - I think this is expected. I've got this working in redshell, I'll know for sure once its working in Kokoro.

File: ci/setup_tpm12_simulator.sh (hunk 2 above)
Comment (twitchy-jsonp) [reply ci/setup_tpm12_simulator.sh]: Done.

**Pull request merged**